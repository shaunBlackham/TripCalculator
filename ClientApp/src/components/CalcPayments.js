import React, { Component } from 'react';
import image1 from '../images/Snow2.PNG';
import image2 from '../images/Tree1.jpeg';
import image3 from '../images/Snow4.JPG';
import { toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

toast.configure() 

function notify() {
    toast.warning("Please navigate back to the Student Entry page and enter at least three students and expenses.", { position: toast.POSITION.TOP_LEFT, autoClose: 5000 });
}

export class CalcPayments extends Component {
    static displayName = CalcPayments.name;

    constructor(props) {
        super(props);
        this.state = { students: [], loading: true };
    }

    componentDidMount() {
        this.populateCalcPayments();
    }

    static renderCalcPaymentsTable(students) {
        return (
            <div>
                <table className='table table-striped' aria-labelledby="tabelLabel">
                    <thead>
                        <tr>
                            <th>Student Name</th>
                            <th>Reimbursement</th>
                            <th>Paid To</th>
                        </tr>
                    </thead>
                    <tbody>
                        {students.map(students =>
                            <tr key={students.name}>
                                <td>{students.name}</td>
                                <td>{students.reimbursement}</td>
                                <td>{students.paidTo}</td>
                            </tr>
                        )}
                    </tbody>
                </table>
                <img src={image1} style={{ height: 300 }} alt="" /><img src={image2} style={{ height: 300 }} alt="" /><img src={image3} style={{ height: 300 }} alt="" />
            </div>
        );
    }

    render() {
        let contents = this.state.loading
            ? <p><em>Loading...</em></p>
            : CalcPayments.renderCalcPaymentsTable(this.state.students);

        return (
            <div>
                <h1 id="tabelLabel" >Who owes who money?</h1>
                {contents}
            </div>
        );
    }

    async populateCalcPayments() {
        const count = await(this.getStudentCount());
        if (count < 3) {
            notify();
        } else {
            const response = await fetch('student');
            const data = await response.json();
            this.setState({ students: data, loading: false });
        }
    }

    async getStudentCount() {
        const requestOptions = {
            method: 'GET',
            headers: { 'Content-Type': 'application/json' }
        }
        const response = await fetch('student/count', requestOptions);
        const data = await response.json();
        return data;
    }
}
