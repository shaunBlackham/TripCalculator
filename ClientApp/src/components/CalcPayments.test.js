import React, { Component } from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import { CalcPayments } from './CalcPayments';

Enzyme.configure({ adapter: new Adapter() });

describe("CalcPayments", () => {
    
    it("should render initial layout", () => {
        // when
        const component = shallow(<CalcPayments />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });

    it('renders three images', () => {
        const component = shallow(<CalcPayments />);
        const images = component.find('img');
        expect(images.length == 3);
    });
});