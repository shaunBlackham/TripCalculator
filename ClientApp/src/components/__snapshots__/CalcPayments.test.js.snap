// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcPayments should render initial layout 1`] = `
Array [
  <div>
    <h1
      id="tabelLabel"
    >
      Who owes who money?
    </h1>
    <p>
      <em>
        Loading...
      </em>
    </p>
  </div>,
]
`;
